src/layout/mod.rs

```diff
@@ -0,0 +1,569 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+#![allow(clippy::module_inception)]
```

Comment:
What about instead of bundling everything in `layout.rs` if we split up the individual layout methods into separate modules like `bipartite.rs`, `spring.rs`, etc. Then we wouldn't need this clippy allow. Also it would give us a place to put `_spring_layout()` and `_random_layout()` to simplify the mod.rs to just be the pyfunctions.

Reply:
Done, now there are six files with the layout algorithms and `mod.rs` is mostly just docstrings